build_deploy.sh

```diff
@@ -29,49 +58,74 @@ initialize_container_engine_cmd() {
         CONTAINER_ENGINE_CMD='podman'
     else
         mkdir -p "$DOCKER_CONF"
-        CONTAINER_ENGINE_CMD="docker --config=${DOCKER_CONF}"
+        CONTAINER_ENGINE_CMD="docker"
+    fi
+}
+
+initialize_image_config() {
+
+    login_container_registry "$RH_REGISTRY_USER" "$RH_REGISTRY_TOKEN" "$REDHAT_REGISTRY"
+
+    if local_build; then
+        IMAGE_NAME="$IMAGE_REPOSITORY"
+    else
+        IMAGE_NAME="$IMAGE_REGISTRY/$IMAGE_REPOSITORY"
+        login_container_registry "$QUAY_USER" "$QUAY_TOKEN" "$IMAGE_REGISTRY"
     fi
 }
 
 build_image() {
 
-    "$CONTAINER_ENGINE_CMD" build -f "$DOCKERFILE" -t "${IMAGE_NAME}:${IMAGE_TAG}" .
+    container_engine_cmd build -f "$DOCKERFILE" -t "${IMAGE_NAME}:${IMAGE_TAG}" .
 }
 
 push_image() {
 
     local IMAGE_TAG="$1"
 
-    "$CONTAINER_ENGINE_CMD" push "${IMAGE_NAME}:${IMAGE_TAG}"
+    container_engine_cmd push "${IMAGE_NAME}:${IMAGE_TAG}"
 }
 
 tag_image() {
 
     local TARGET_TAG="$1"
 
-    "$CONTAINER_ENGINE_CMD" tag "${IMAGE_NAME}:${IMAGE_TAG}" "${IMAGE_NAME}:$TARGET_TAG"
+    container_engine_cmd tag "${IMAGE_NAME}:${IMAGE_TAG}" "${IMAGE_NAME}:$TARGET_TAG"
 }
 
 tag_and_push_for_backwards_compatibility() {
 
     for TARGET_TAG in "latest-new" "qa-new"; do
         tag_image "$TARGET_TAG"
-        push_image "$TARGET_TAG"
+        if ! local_build; then
+            push_image "$TARGET_TAG"
+        fi
     done
 }
 
-IMAGE_NAME="quay.io/cloudservices/compliance-backend"
-IMAGE_TAG="$(git rev-parse --short=7 HEAD)-new"
-WORKDIR="$PWD"
+if ! local_build; then
+    set -xv
+fi
```

Comment:
This doesn't work the same way now when local build is false. It needs to be at the top of the file if you want all the function definitions to be printed to STDOUT.

Reply:
reverted it back and left the flags as they were